backend/app/verification.py

```diff
@@ -884,6 +912,78 @@ def unverify(
         worker.republish_app.send(app_id)
 
 
+@router.post("/{app_id}/enroll", status_code=204, tags=["verification"])
+def enroll(
+    login=Depends(logged_in),
+    app_id: str = Path(
+        min_length=6,
+        max_length=255,
+        pattern=r"^[A-Za-z_][\w\-\.]+$",
+        examples=["org.gnome.Glade"],
+    ),
+):
+    is_direct_upload = models.DirectUploadApp.by_app_id(sqldb, app_id) is not None
+
+    if is_direct_upload:
+        return
+
+    is_verified = _get_existing_verification(app_id)
+    if is_verified and not is_direct_upload:
+        _create_direct_upload_app(login.user, app_id)
+        _archive_github_repo(app_id)
+
+
+@router.post("/{app_id}/archive", status_code=204, tags=["verification"])
+def archive(
+    request: ArchiveRequest,
+    login=Depends(logged_in),
+    app_id: str = Path(
+        min_length=6,
+        max_length=255,
+        pattern=r"^[A-Za-z_][\w\-\.]+$",
+        examples=["org.gnome.Glade"],
+    ),
+):
+    if not config.settings.flat_manager_api:
+        return
+
+    direct_upload_app = models.DirectUploadApp.by_app_id(sqldb, app_id)
+    if direct_upload_app is not None:
+        if direct_upload_app.archived:
+            return
+
+        direct_upload_app.archived = True
+        sqldb.session.commit()
+
+    if not direct_upload_app:
+        gh_repo_changed = _archive_github_repo(app_id)
+        if not gh_repo_changed:
+            return
+
+    if app_id not in login.user.dev_flatpaks(sqldb):
+        raise HTTPException(
+            status_code=403,
+            detail=ErrorDetail.NOT_APP_DEVELOPER,
+        )
+    upload_tokens = models.UploadToken.by_app_id(sqldb, app_id)
+    flat_manager_jwt = utils.create_flat_manager_token(
+        "revoke_upload_token", ["tokenmanagement"], sub=""
+    )
+    for token in upload_tokens:
+        response = requests.post(
+            config.settings.flat_manager_api + "/api/v1/tokens/revoke",
+            headers={"Authorization": flat_manager_jwt},
+            json={"token_ids": [_jti(token)]},
+        )
+
+        if not response.ok:
+            raise HTTPException(status_code=500)
```

Comment:
A bit worried about this and other return paths, as you potentially exit early and can't do this again - as you've already commited database things above and that will block you from rerunning.

Or one token failing to be revoked would lead to no republish

Reply:
I think I need to push this to the worker, and we should also wrap all external requests to retry on failure. Will reshuffle the code but will address this in another PR.